test/integration/multinode_test.go

```diff
@@ -387,3 +408,44 @@ func validatNameConflict(ctx context.Context, t *testing.T, profile string) {
 		t.Logf("failed to clean temporary profile. args %q : %v", rr.Command(), err)
 	}
 }
+
+func validateDeployAppToMultiNode(ctx context.Context, t *testing.T, profile string) {
+	// Create a deployment for app
+	_, err := Run(t, exec.CommandContext(ctx, Target(), "kubectl", "-p", profile, "--", "apply", "-f", "./testdata/multinodes/hello-deployment.yaml"))
+	if err != nil {
+		t.Errorf("failed to create hello deployment to multinode cluster")
+	}
+
+	_, err = Run(t, exec.CommandContext(ctx, Target(), "kubectl", "-p", profile, "--", "rollout", "status", "deployment/hello"))
+	if err != nil {
+		t.Errorf("failed to delploy hello to multinode cluster")
+	}
+
+	// resolve Pod IPs
+	rr, err := Run(t, exec.CommandContext(ctx, Target(), "kubectl", "-p", profile, "--", "get", "pods", "-o", "jsonpath='{.items[*].status.podIP}'"))
+	if err != nil {
+		t.Errorf("failed retrieve Pod IPs")
+	}
+	podIPs := strings.Split(strings.Trim(rr.Stdout.String(), "'"), " ")
+	if len(podIPs) != 2 {
+		t.Errorf("expected 2 Pod IPs but got %d", len(podIPs))
+	}
+	if podIPs[0] == podIPs[1] {
+		t.Errorf("expected 2 different pod IPs but got %s and %s", podIPs[0], podIPs[0])
+	}
```

Comment:
If possible verify pods in node are in the the `node.PodCIDR` range.

Reply:
Why we need this check?